lib/block_repeater/repeater.rb

```diff
@@ -72,6 +74,43 @@ def repeat(times: 25, delay: 0.2, **_)
       result
     end
 
+    # Retry a call whilst exponentially increasing the wait time between each iteration until a timeout is reached
+    # @param [Integer] `timeout` is the max amount wait of time you wish to try the action
+    # @param [Integer] `initial_wait` is the initial wait time to retry the action
+    # @param [Integer] `multiplier` is the rate at which you increase the wait time between each iteration
+    def backoff(timeout: 10, initial_wait: 0.1, multiplier: 2)
+      raise StandardError, 'Multiplier cannot be less than 1.1' if multiplier < 1.1
+
+      condition_met = nil
+      result = nil
+
+      current_sleep_seconds = initial_wait
+      start_time = Time.now
+
+      until current_sleep_seconds >= timeout
+        result = @repeat_block.call
+        condition_met = @condition_block.call(result) if @condition_block
+
+        duration = Time.now - start_time
+
+        if duration > timeout
+          LOGGER.warn "Timeout reached on fallback - Duration: #{duration} - Timeout: #{timeout} seconds"
```

Comment:
```suggestion
          LOGGER.debug "Timeout reached on fallback - Duration: #{duration} - Timeout: #{timeout} seconds"
```

This is more a debug or info statement. Warn levels, to me, are something bad happened that you need to be aware of